Tools/CISupport/ews-build/steps.py

```diff
@@ -2150,6 +2153,189 @@ def hideStepIf(self, results, step):
         return not self.doStepIf(step)
 
 
+class FilterPullRequestsIntoQueues(buildstep.BuildStep, AddToLogMixin):
+    name = 'filter-pull-requests-into-queues'
+
+    # @defer.inlineCallbacks TODO: implement twistd
+    def run(self):
+        num_prs = self.getNumPRs()
+        if num_prs:
+            sorted_prs = self.sortPRs(num_prs)
+            if sorted_prs:
+                list_of_prs, label_as_safe, label_as_blocked = sorted_prs
+            else:
+                return FAILURE
+            self.setProperty('safe_pr_list', label_as_safe)
+            self._addToLog('stdio', 'PRs to merge: {}.'.format(label_as_safe))
+            self.setProperty('blocked_pr_list', label_as_blocked)
+            self._addToLog('stdio', 'PRs to block: {}'.format(label_as_blocked))
+
+            if len(label_as_safe):
+                self.build.addStepsAfterCurrentStep([SafeMergeToMergeQueue()])
+            if len(label_as_blocked):
+                self.build.addStepsAfterCurrentStep([SafeMergeToMergeBlocked()])
+        return SUCCESS if num_prs else FAILURE
+
+    def getResultSummary(self):
+        if self.results == SUCCESS:
+            return {'step': f"Successfully requested and filtered pull requests"}
+        elif self.results == FAILURE:
+            return {'step': f"Failed to request and filter pull requests"}
+        return buildstep.BuildStep.getResultSummary(self)
+
+    def queryGraphQL(self, payload):
+        graphql_url = 'https://api.github.com/graphql'
+        username, access_token = GitHub.credentials(user=GitHub.user_for_queue(self.getProperty('buildername', '')))
+        if username and access_token:
+            auth_header = b64encode('{}:{}'.format(username, access_token).encode('utf-8')).decode('utf-8')
+            headers = {'Authorization': 'bearer {}'.format(access_token)}
+
+        # TODO: implement twistd
+        response = requests.post(url=graphql_url, json=payload, headers=headers)
+        data = json.loads(response.text)
+        if 'errors' in response:
+            self._addToLog('stdio', response['errors'])
+        return data
+
+    def getNumPRs(self, label='\"safe-merge-queue\"'):
+        query_body = '{ repository(owner:"Webkit", name:"WebKit") { pullRequests(labels: %s) { totalCount } } }' % label
+        query = {'query': query_body}
+        num_prs = self.queryGraphQL(query)['data']['repository']['pullRequests']['totalCount']
+        if num_prs:
+            yield self._addToLog('stdio', 'There are {} PRs in safe-merge-queue.'.format(num_prs))
```

Comment:
Every function with a `yield` should be decorated with `@defer.inlineCallbacks` and every function invoking a function with `@defer.inlineCallbacks` needs to preface that invocation of with `yield`